### PR TITLE
build: Add nix flake

### PR DESCRIPTION
**Pull request by Skallwar**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  description = "GoOSe devshell";
```
Comment (CohenArthur) [inline flake.nix]: :eyes:

Comment (Skallwar) [reply flake.nix]: :eyes:

Comment (Skallwar) [reply flake.nix]: Fixed

**Review by CohenArthur — APPROVED** [review]
No idea how to review this but I trust you!

**Pull request merged**